engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportSizeOf.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.terasology.rendering.dag.FBOManagerSubscriber;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.dag.tasks.SetViewportSizeOfTask;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+
+/**
+ * TODO: Add javadocs
+ */
+public final class SetViewportSizeOf implements FBOManagerSubscriber, StateChange {
+    private static final String DEFAULT_FBO_NAME = "sceneOpaque";
+    private static SetViewportSizeOf defaultInstance = new SetViewportSizeOf();
+    private SetViewportSizeOfTask task;
+
+    private FBO fbo;
+    private String fboName;
+    private FrameBuffersManager frameBuffersManager;
+
+    public SetViewportSizeOf(String fboName, FrameBuffersManager frameBuffersManager) {
+        this.fboName = fboName;
+        this.frameBuffersManager = frameBuffersManager;
+        fbo = frameBuffersManager.getFBO(fboName);
+    }
+
+    public SetViewportSizeOf() {
+        this.fboName = DEFAULT_FBO_NAME;
+    }
+
+    @Override
+    public StateChange getDefaultInstance() {
```

Comment:
Hmmm. Trickytrickytricky. I have a couple of thoughts.

1. Theoretically, the default instance could reset the viewport to the size of the window rather than the size of a buffer, and this would free the default instance from having to deal with the FBM. This would work if we accept that we need to query the size of the display every frame and monitor for changes. Perhaps a subscribable object could do the monitoring work, notifying subscribers such as the default instance if display changes.

2. In practice, how about you convert the frameBuffersManager variable into a static one, and you initialize it in the WorldRendererImpl, just after the frameBuffersManager is instantiated? This way the class will always have the instance of the FBM available and you can even shorten the constructors.